daemons/dss-sync/app.py

```diff
@@ -12,7 +14,11 @@
 
 @app.s3_event_handler(bucket=s3_bucket, events=["s3:ObjectCreated:*"])
 def process_new_syncable_object(event, context):
-    s3 = boto3.resource("s3")
-    bucket = s3.Bucket(event['Records'][0]["s3"]["bucket"]["name"])
-    obj = bucket.Object(unquote(event['Records'][0]["s3"]["object"]["key"]))
-    sync_blob(source_platform="s3", source_key=obj.key, dest_platform="gcs", logger=app.log)
+    app.log.setLevel(logging.DEBUG)
+    if event.get("Event") == "s3:TestEvent":
+        app.log.info("DSS sync daemon received S3 test event")
+    else:
+        s3 = boto3.resource("s3")
+        bucket = s3.Bucket(event['Records'][0]["s3"]["bucket"]["name"])
+        obj = bucket.Object(unquote(event['Records'][0]["s3"]["object"]["key"]))
```

Comment:
It doesn't hurt. Even though we moved away from colons, S3 keys can generally be named almost anything. They get URI-encoded in the event notification, so we should decode them before we can use them.